src/CfgMgr32/CfgMgr32+CM_NOTIFY_EVENT_DATA.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="CM_NOTIFY_EVENT_DATA"/> nested struct.
+    /// </content>
+    public static partial class CfgMgr32
+    {
+        /// <summary>
+        /// This is a device notification event data structure.
+        /// </summary>
+        /// <seealso hcref="https://docs.microsoft.com/en-us/windows/win32/api/cfgmgr32/ns-cfgmgr32-cm_notify_event_data"/>
+        [StructLayout(LayoutKind.Explicit)]
+        public unsafe struct CM_NOTIFY_EVENT_DATA
```

Comment:
Should we use [`ref struct`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#ref-struct) here so that a bunch of restrictions apply, increasing the likelihood that in memory, the data *after* the struct remains the trailing `char` array that we expect it to be?